hw_diag/utilities/balena_supervisor.py

```diff
@@ -43,17 +44,23 @@ def shutdown(self):
         response = self._make_request('POST', '/v1/shutdown')
         if response is None or response.ok is False:
             LOGGER.error("Device shutdown attempt failed.")
+            raise RuntimeError('supervisor API not accessible')
 
-        return response.json()
+        try:
+            return response.json()
+        except Exception:
+            raise RuntimeError('shutdown failed due to supervisor API issue')
 
     def get_device_status(self, key_to_return) -> str:
         """Get device status from balena supervisor API."""
         LOGGER.info("Retrieving device status using Balena supervisor.")
 
         response = self._make_request('GET', '/v2/state/status')
-        LOGGER.info(response.json())
         if response is None or response.ok is False:
             LOGGER.error("Device status request failed.")
             raise Exception("Device status request failed")
 
-        return response.json()[key_to_return]
+        try:
+            return response.json()[key_to_return]
+        except Exception:
+            return 'Failed due to supervisor API issue'
```

Comment:
@kashifpk why return a string here instead of raising an exception?

Reply:
> @kashifpk why return a string here instead of raising an exception?

@marvinmarnold - That's a mistake at my end thinking that we'll just display this message as device status as the return value is expected to be a string. However this won't cause the device diagnostic to fail which is incorrect behavior. I'll open a new PR and correct this.